### DISCUSSION TITLE
C# Task dose not trigger the SimulationFinished

### DISCUSSION THREAD

**Discussion by Abbas111977** [SpiceSharp/SpiceSharp#185] [General]
Hello, I have used this library with a list of Task in C#, but it dose not trigger the SimulationFinished at all, and I am wondering if there is any solution to this problem.
Note: it would be very kind from you if the solution presented as Code example.
Best Regards

**Comment by svenboulanger** [comment]
Can you please elaborate? There is no `SimulationFinished` event in Spice#. The last event that should get called is the `AfterUnsetup` event.

**Reply by Abbas111977** [reply]
Sorry it was my fault, and i have solve it now, but i need your help regarding ideal diode implementation example, i have tried the following but it didn't work:  new Diode("D1", "out", "in", ".model Didl D(Ron=0.0001 Roff=1G Vfwd=.6)")
Thanks for your replay,
Kind Regards

**Reply by marcin-golebiowski** [reply]
@Abbas111977  Take a look at: https://github.com/SpiceSharp/SpiceSharp/blob/master/SpiceSharpTest/Models/Semiconductors/DIO/DiodeTests.cs

**Reply by Abbas111977** [reply]
thank you very much
